presentation/src/main/java/com/example/healthysalad/ViewAdapter.kt

```diff
@@ -42,5 +42,10 @@ class MyAdapter(private val myDataset: List<Base>?) :
     }
 
     // Return the size of your dataset (invoked by the layout manager)
-//    override fun getItemCount(): Int = myDataset?.size!!
+    // override fun getItemCount(): Int = myDataset?.size!!
+    fun updateMyDataset(list: List<Base>) {
+        myDataset.clear()
+        myDataset.addAll(list)
+        notifyItemInserted(0)
```

Comment:
You set a whole salad data for the recycler view item, not insert just a part of the data.

If you read the [notifyItemInserted document](https://developer.android.com/reference/androidx/recyclerview/widget/RecyclerView.Adapter#notifyItemInserted(int)), you can think that in this case, [notifyDataSetChanged()](https://developer.android.com/reference/androidx/recyclerview/widget/RecyclerView.Adapter#notifyDataSetChanged()) is a better one.